edgelet/edgelet-http/src/lib.rs

```diff
@@ -257,3 +262,34 @@ impl HyperExt for Http {
         })
     }
 }
+
+pub trait UrlExt {
+    fn to_uds_file_path(&self) -> Result<PathBuf, Error>;
+}
+
+impl UrlExt for Url {
+    #[cfg(unix)]
+    fn to_uds_file_path(&self) -> Result<PathBuf, Error> {
+        debug_assert_eq!(self.scheme(), UNIX_SCHEME);
+        self.to_file_path()
+            .map_err(|()| ErrorKind::InvalidUri(self.to_string()).into())
+    }
+
+    #[cfg(windows)]
+    fn to_uds_file_path(&self) -> Result<PathBuf, Error> {
+        // We get better handling of Windows file syntax if we parse a
+        // unix:// URL as a file:// URL. Specifically:
+        // - On Unix, `Url::parse("unix:///path")?.to_file_path()` succeeds and
+        //   returns "/path".
+        // - On Windows, `Url::parse("unix:///C:/path")?.to_file_path()` fails
+        //   with Err(()).
+        // - On Windows, `Url::parse("file:///C:/path")?.to_file_path()` succeeds
+        //   and returns "C:\\path".
+        debug_assert_eq!(self.scheme(), UNIX_SCHEME);
```

Comment:
This seems to work on Windows, and avoids the need to assert the scheme and replace the scheme via string replace.

```rust
let mut url = self.clone();
url.set_scheme("file").map_err(|()| ErrorKind::InvalidUri(self.to_string()).into())?;
url.to_file_path().map_err(|()| ErrorKind::InvalidUri(self.to_string()).into())
```

Reply:
I (thought I) tried `set_scheme` and it didn't help. I'll definitely investigate this again...would love to get rid of the workaround.

Reply:
Never mind, you're right.

For `unix://C:/path` + `set_scheme` it succeeds but gives `\\\\C\\path`

For `unix:///C:/path` + `set_scheme` it fails with `to_file_path() failed to produce an absolute Path`